src/hotspot/share/opto/addnode.cpp

```diff
@@ -790,6 +808,13 @@ Node* OrINode::Ideal(PhaseGVN* phase, bool can_reshape) {
       return new RotateRightNode(in(1)->in(1), shift, TypeInt::INT);
     }
   }
+
+  // Convert "~a | ~b" into "~(a & b)"
+  if (is_not(phase, in(1), T_INT) && is_not(phase, in(2), T_INT)) {
+    return make_not(phase,
+                    phase->transform(new AndINode(in(1)->in(1), in(2)->in(1))),
+                    T_INT);
```

Comment:
I'd put the `AndI` node on a separate line. Call it `add_a_b` or similar.
Then you can transform on the next line.
And then on a third line the `make_not`.